### PR TITLE
Switch test site from jest to vitest

### LINKED ISSUE #208: Switch Test Suite from Jest to Vitest
Relates to:
- https://vitest.dev/guide/
- https://github.com/icefoganalytics/internal-data-portal/issues/98
- https://github.com/icefoganalytics/elcc-data-management/pull/77
- https://github.com/icefoganalytics/wrap/pull/33
- https://github.com/icefoganalytics/wrap/pull/32

# Context

**Is your feature request related to a problem? Please describe.**
Jest is slow, the docs are hard to read, and the behaviour doesn't play well with Sequelize 7. Vitest is fast, the docs are easy to read, and it works well with Sequelize 7. It even has full Jest expectation support, so the transition is quite smooth.

Most importanly! Vitest has precise file watching, so that if you edit a file in watch mode, it only runs the tests that effect that file. This is a vast UX improvement over Jest which requires to you manually specify only watching a single file, each time, or to run the whole suite.

**Describe the solution you'd like**
Switch our test suite from Jest to Vitest.

**Describe alternatives you've considered**
Keep using Jest - it works ok for now, not great, but ok.
Switch to Mocha + TS-Mocha + Chai - much harder migration, with no clear gain over switching to Vitest, which has both Chai and Jest support.

### PR DESCRIPTION
**Pull request by burkkyy**
Fixes #208 

Relates to:
- https://vitest.dev/guide/
- https://github.com/icefoganalytics/internal-data-portal/issues/98
- https://github.com/icefoganalytics/elcc-data-management/pull/77
- https://github.com/icefoganalytics/wrap/pull/33
- https://github.com/icefoganalytics/wrap/pull/32

# Context

**Is your feature request related to a problem? Please describe.**
Jest is slow, the docs are hard to read, and the behaviour doesn't play well with Sequelize 7. Vitest is fast, the docs are easy to read, and it works well with Sequelize 7. It even has full Jest expectation support, so the transition is quite smooth.

Most importanly! Vitest has precise file watching, so that if you edit a file in watch mode, it only runs the tests that effect that file. This is a vast UX improvement over Jest which requires to you manually specify only watching a single file, each time, or to run the whole suite.

**Describe the solution you'd like**
Switch our test suite from Jest to Vitest.

**Describe alternatives you've considered**
Keep using Jest - it works ok for now, not great, but ok.
Switch to Mocha + TS-Mocha + Chai - much harder migration, with no clear gain over switching to Vitest, which has both Chai and Jest support.

# Implementation

1. Switch from jest to vitest.
2. Remove Jest + Sequelize error reporting monkey patch.
3. Upgrade initialization runner using new code from [WRAP](https://github.com/icefoganalytics/wrap/blob/fab2ba7e252f04285eee27a4e42ba9541089d121/api/src/initializers/index.ts)
4. Upgrade request mocking capabilities with more code from WRAP.
5. Increase tsconfig target and module type to max available without switching to ESM (package type module)

# Screenshots
image

## Testing Instructions
1. Run `dev down`
2. Run the test suite via `dev test`

### CODE REVIEW

File: api/development.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:16-alpine3.15
```
Comment (klondikemarlen) [inline api/development.Dockerfile:1]: You'll probably need to make this change in a separate PR as it will likely break a tonne of stuff. Also be aware that this must match the production `Dockerfile` at all times, or the app will no longer build.

Comment (burkkyy) [reply api/development.Dockerfile:1]: Submitted PR to update node version in #211
